File: scripts/export/v0_33_0/main.go
package main

import (
	"flag"
	"fmt"

	app "github.com/cosmos/cosmos-sdk/cmd/gaia/app"
	"github.com/cosmos/cosmos-sdk/scripts/export"
)

const chainID = "cosmoshub-2"

// Command: go run main.go --path=<path_to_old_genesis.json> --start-time=<genesis-start-time> > [path_to_new_genesis.json
func main() {
	cdc := app.MakeCodec()

	pathToGenesis := flag.String("path", "", "./genesis.json")
	genesisTime := flag.String("start-time", "", "2019-02-11T12:00:00Z")

	flag.Parse()

	err := export.ValidateBasic(*pathToGenesis, *genesisTime)
	if err != nil {
		panic(err)
	}

	genesis, err := export.NewGenesisFile(cdc, *pathToGenesis)
	if err != nil {
		panic(err)
	}

	genesis.ChainID = chainID
	genesis.GenesisTime = *genesisTime

	// proposal #1 updates
	genesis.AppState.MintData.Params.BlocksPerYear = 4855015

	// proposal #2 updates
	genesis.ConsensusParams.Block.MaxGas = 200000
	genesis.ConsensusParams.Block.MaxBytes = 2000000

	// enable transfers
	genesis.AppState.BankData.SendEnabled = true
	genesis.AppState.DistrData.WithdrawAddrEnabled = true

	err = app.GaiaValidateGenesisState(genesis.AppState)
	if err != nil {
		panic(err)
	}

	genesisJSON, err := cdc.MarshalJSONIndent(genesis, "", "  ")
	if err != nil {
		panic(err)
	}
	fmt.Println(string(genesisJSON))
}

File: scripts/export/v0_33_0/main.py
#!/usr/bin/env python3

import argparse
import datetime
import json
import math
import sys


# default_date = datetime.datetime(year=2019, month=2, day=11, hour=12, tzinfo=datetime.timezone.utc) # '2019-02-11T12:00:00Z'

parser = argparse.ArgumentParser(description='Convert genesis.json from v0.33.2 to v0.34.0')
parser.add_argument('exported_genesis', type=argparse.FileType('r'), default=sys.stdin)
parser.add_argument('--chain-id', type=str, default='cosmoshub-n')
parser.add_argument('--start-time', type=str, default='2019-02-11T12:00:00Z')
args = parser.parse_args()

chain_id = args.chain_id.strip()

if chain_id == '':
  sys.exit('chain-id required')

raw = args.exported_genesis.read()
genesis = json.loads(raw)

# update genesis with breaking changes
genesis['consensus_params']['block'] = genesis['consensus_params']['block_size']
del genesis['consensus_params']['block_size']
genesis['consensus_params']['block']['time_iota_ms'] = '1000' # default tm value


crisis_data = {
  'constant_fee': {
    'amount': '1333000000',  # $5,000 worth of uatoms
    'denom': 'uatom'
  }
}
genesis['app_state']['crisis'] = crisis_data

# proposal #1 updates
genesis['app_state']['mint']['params']['blocks_per_year'] = '4855015'

# proposal #2 updates
genesis['consensus_params']['block']['max_gas'] = '2000000'
genesis['consensus_params']['block']['max_bytes'] = '200000'

# enable transfers
genesis['app_state']['bank']['send_enabled'] = True
genesis['app_state']['distr']['withdraw_add_enabled'] = True

# Set new chain ID and genesis start time 
genesis['chain_id'] = chain_id.strip()
genesis['genesis_time'] = args.start_time

print(json.dumps(genesis, indent=True))
